File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
n = int(input())

if n%5 == 1:
    if n == 1:
        print(-1)
    else:
        print(n//5+1)
elif n%5 == 2:
    if n ==2 or n ==7:
        print(-1)
    else:
        print(n//5+2)
elif n%5 ==3:
    print(n//5+1)
elif n%5 ==4:
    if n==4:
        print(-1)
    else:
        print(n//5+2)
else:
    print(n//5)

File: 백준/Silver/2839. 설탕 배달/설탕 배달.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    static BufferedReader input = new BufferedReader(new InputStreamReader(System.in));

    public static void main(String[] args) throws IOException {
        int n = Integer.parseInt(input.readLine());
        if (n%5 == 1){
            if (n==1){
                System.out.println(-1);
            }else{
                System.out.println(n/5+1);
            }
        } else if(n%5 == 2){
            if (n==2 || n==7){
                System.out.println(-1);
            }else{
                System.out.println(n/5+2);
            }
        } else if (n%5 == 3) {
            System.out.println(n/5+1);
        } else if (n%5 == 4){
            if (n==4){
                System.out.println(-1);
            }else{
                System.out.println(n/5+2);
            }
        } else{
            System.out.println(n/5);
        }

    }
}